rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,212 @@
+                let tick_size = opt_value_to_big_decimal(event.data.get("tick_size"));
+                let min_size = opt_value_to_big_decimal(event.data.get("min_size"));
+                let underwriter_id = opt_value_to_big_decimal(event.data.get("underwriter_id"));
+                let time = *block_height_to_timestamp
+                    .get(&event.transaction_block_height)
+                    .unwrap();
+                let base_name_generic;
+                let base_account_address;
+                let base_module_name;
+                let base_struct_name;
+                if value_is_empty_string(&event.data["base_name_generic"]) {
+                    base_name_generic = None;
+                    base_account_address = Some(String::from(
+                        // cannot panic because account_address always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["account_address"].as_str().unwrap(),
+                    ));
+                    let base_module_name_hex =
+                        // cannot panic because module_name always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["module_name"].as_str().unwrap();
+                    let base_struct_name_hex =
+                        // cannot panic because struct_name always exists as a string
+                        // assuming base_name_generic was empty
+                        event.data["base_type"]["struct_name"].as_str().unwrap();
+                    base_module_name = Some(hex_to_string(base_module_name_hex));
+                    base_struct_name = Some(hex_to_string(base_struct_name_hex));
+                } else {
+                    base_name_generic = Some(String::from(
+                        // cannot panic because base_name_generic always exists as a string
+                        event.data["base_name_generic"].as_str().unwrap(),
+                    ));
+                    base_account_address = None;
+                    base_module_name = None;
+                    base_struct_name = None;
+                }
+                let quote_account_address = String::from(
+                    event.data["quote_type"]["account_address"]
+                        .as_str()
+                        .unwrap(),
+                );
+                let quote_module_name_hex =
+                    // cannot panic because module_name always exists as a string
+                    event.data["quote_type"]["module_name"].as_str().unwrap();
+                let quote_struct_name_hex =
+                    // cannot panic because struct_name always exists as a string
+                    event.data["quote_type"]["struct_name"].as_str().unwrap();
+                let quote_module_name = hex_to_string(quote_module_name_hex);
+                let quote_struct_name = hex_to_string(quote_struct_name_hex);
```

Comment:
I would again replace `unwrap` with `?`.

Reply:
These are options sadly, not results